File: leetcode-cpp/1200_1299/leetcode-no1295.cpp
#include <iostream>
#include <string>
#include <vector>

using namespace std;
class Solution {
public:
    int findNumbers(vector<int>& nums) {
        int ans = 0;
        for (auto k : nums) {
            if (not (to_string(k).size() % 2)){
//                cout << to_string(k).size() << " " << (to_string(k).size() % 2) << " " << endl;
                ans ++;
            }
        }
        return ans;
    }
};

int main(){
    Solution test;
    vector<int> eg1 = {12,345,2,6,7896};
    cout << test.findNumbers(eg1) << endl;
}
//
// Created by ASUS on 2023/7/15.
//

File: leetcode-rs/1200_1299/leetcode-no1295.rs
// problem: https://leetcode.cn/problems/find-numbers-with-even-number-of-digits/?envType=daily-question&envId=2025-04-30
struct Solution;

impl Solution {
    pub fn find_numbers(nums: Vec<i32>) -> i32 {
        let mut res = 0;
        for mut num in nums {
            let mut times = 0;
            while num > 0 {
                num /= 10;
                times += 1;
            }
            if times % 2 == 0 {
                res += 1;
            }
        }
        res
    }
}

fn main() {
    println!("{:?}", Solution::find_numbers(vec![12,345,2,6,7896]));
    println!("{:?}", Solution::find_numbers(vec![555,901,482,1771]));
}

//
// Created by zhiya at 2025/4/30 14:14
//